client/core/core.go

```diff
@@ -2102,9 +2102,9 @@ func (c *Core) updateBalances(assets assetMap) {
 	for assetID := range assets {
 		w, exists := c.wallet(assetID)
 		if !exists {
-			// This should never be the case, but log an error in case I'm
-			// wrong or something changes.
-			c.log.Errorf("non-existent %d wallet should exist", assetID)
+			// Could happen if we are here when handling "tip change" before Core
+			// managed to initialize its wallets map (or removed this wallet already).
+			c.log.Tracef("can't update %q balance, Core isn't tracking this wallet yet/still", unbip(assetID))
```

Comment:
I'd like to better understand the circumstances under which this can happen? It doesn't see like it's possible during core initialization. Does this happen during new wallet creation?

Reply:
Can also happen on shutdown it seems:

```
2023-03-29 02:44:39.572 [INF] CORE[dcr][SPV]: Unloading wallet
2023-03-29 02:44:39.573 [INF] CORE[dcr][SPV]: SPV wallet closed
2023-03-29 02:44:39.573 [INF] CORE: Locking ETH wallet
2023-03-29 02:44:39.633 [INF] CORE[eth][TXDB]: Got compaction priority: {level:0 score:1.73 dropPrefixes:[]}
2023-03-29 02:44:39.634 [INF] CORE: Locking USDC.ETH wallet
2023-03-29 02:44:39.634 [ERR] CORE: Failed to lock USDC.ETH wallet: wallet not connected
2023-03-29 02:44:39.634 [INF] CORE: DEX client core off
2023-03-29 02:44:39.634 [INF] DEXC: Exiting dexc main.
2023-03-29 02:44:39.634 [ERR] CORE: non-existent 60 wallet should exist
2023-03-29 02:44:39.634 [ERR] CORE: non-existent 60001 wallet should exist
2023-03-29 02:44:39.634 [ERR] CORE: non-existent 2 wallet should exist
2023-03-29 02:44:39.634 [ERR] CORE: non-existent 20 wallet should exist
```